src/pages/Compare/index.spec.tsx

```diff
@@ -76,266 +95,349 @@ describe(__filename, () => {
         addonId,
-        baseVersionId: baseVersion.id,
-        headVersionId: headVersion.id + 10,
+        baseVersionId,
+        headVersionId: headVersionId + 1,
       }),
+      _fetchDiff,
       store,
     });
 
     dispatch.mockClear();
-    _fetchVersion.mockClear();
+    _fetchDiff.mockClear();
     root.setProps({
       match: {
-        params: getRouteParams({
-          addonId,
-          baseVersionId: baseVersion.id,
-          headVersionId: headVersion.id,
-        }),
+        params: getRouteParams({ addonId, baseVersionId, headVersionId }),
       },
     });
 
     expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
-    expect(_fetchVersion).toHaveBeenCalledWith({
+    expect(_fetchDiff).toHaveBeenCalledWith({
       addonId,
-      versionId: baseVersion.id,
+      baseVersionId,
+      headVersionId,
     });
   });
 
-  it('dispatches fetchVersion() on update if addon ID is different', () => {
+  it('dispatches fetchDiff() on update if addon ID is different', () => {
     const addonId = 123456;
-    const baseVersion = fakeVersion;
+    const baseVersionId = 1;
+    const headVersionId = baseVersionId + 1;
 
     const store = configureStore();
     const fakeThunk = createFakeThunk();
-    const _fetchVersion = fakeThunk.createThunk;
+    const _fetchDiff = fakeThunk.createThunk;
     const dispatch = spyOn(store, 'dispatch');
 
     const root = render({
-      _fetchVersion,
+      _fetchDiff,
       ...getRouteParams({
         addonId: addonId + 10,
-        baseVersionId: baseVersion.id,
       }),
       store,
     });
 
     dispatch.mockClear();
-    _fetchVersion.mockClear();
+    _fetchDiff.mockClear();
     root.setProps({
       match: {
-        params: getRouteParams({
-          addonId,
-          baseVersionId: baseVersion.id,
-        }),
+        params: getRouteParams({ addonId, baseVersionId, headVersionId }),
       },
     });
 
     expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
-    expect(_fetchVersion).toHaveBeenCalledWith({
+    expect(_fetchDiff).toHaveBeenCalledWith({
       addonId,
-      versionId: baseVersion.id,
+      baseVersionId,
+      headVersionId,
     });
   });
+
+  it('dispatches updateSelectedPath() when a file is selected', () => {
+    const version = fakeVersionWithDiff;
+    const headVersionId = version.id;
+    const path = 'new-path.js';
+
+    const store = configureStore();
+    _loadDiff({ headVersionId, store, version });
+
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({ ...getRouteParams({ headVersionId }), store });
+
+    dispatch.mockClear();
+
+    const onSelectFile = root.find(FileTree).prop('onSelect');
+    onSelectFile(path);
+
+    expect(dispatch).toHaveBeenCalledWith(
+      versionsActions.updateSelectedPath({
+        selectedPath: path,
+        versionId: headVersionId,
+      }),
+    );
+  });
+
+  it('does not dispatch fetchDiff() twice for the default file', () => {
```

Comment:
This test is good but I don't see coverage of the path equality check. All tests pass when I make this change but at least one test should fail:

```diff
diff --git a/src/pages/Compare/index.tsx b/src/pages/Compare/index.tsx
index 0e33d7a..d519776 100644
--- a/src/pages/Compare/index.tsx
+++ b/src/pages/Compare/index.tsx
@@ -75,7 +75,7 @@ export class CompareBase extends React.Component<Props> {
 
     if (
       !prevProps ||
-      (prevProps.path && path !== prevProps.path) ||
+      (prevProps.path) ||
       addonId !== prevProps.match.params.addonId ||
       baseVersionId !== prevProps.match.params.baseVersionId ||
       headVersionId !== prevProps.match.params.headVersionId
```